did_x509/did_x509.go

```diff
@@ -49,9 +51,28 @@ func CreateDid(signingCert, caCert *x509.Certificate, subjectAttributes []x509_c
 	formattedDid, err := FormatDid(caCert, policies...)
 	return formattedDid, err
 }
+
+// PercentEncode encodes a string using percent encoding.
+// we can not use url.PathEscape because it does not escape : $ & + = : @ characters.
+// See https://github.com/golang/go/issues/27559#issuecomment-449652574
+func PercentEncode(input string) string {
```

Comment:
at line 47, there's an ignored `err` variable. Maybe you can check it, while you're at it?